packages/runtime-vapor/src/apiCreateIf.ts

```diff
@@ -2,17 +2,15 @@ import { type BlockFn, DynamicFragment } from './block'
 import { renderEffect } from './renderEffect'
 
 export function createIf(
-  condition: () => any,
-  b1: BlockFn,
-  b2?: BlockFn,
+  ifBlockGetter: () => BlockFn | undefined,
   once?: boolean,
   // hydrationNode?: Node,
 ): DynamicFragment {
   const frag = __DEV__ ? new DynamicFragment('if') : new DynamicFragment()
   if (once) {
-    frag.update(condition() ? b1 : b2)
+    frag.update(ifBlockGetter())
   } else {
-    renderEffect(() => frag.update(condition() ? b1 : b2))
+    renderEffect(() => frag.update(ifBlockGetter()))
   }
   return frag
 }
```

Comment:
Maybe we can use a table-driven approach. Add a feedback parameter, and change the ifBlockGetter part to get an array.

Like this:

```typescript
export function createIf(
  ifBlockGetter: [when: () => any, block: BlockFn][]
  feedback?: BlockFn,
  once?: boolean
)
```